src/coreclr/src/jit/block.h

```diff
@@ -465,7 +465,7 @@ struct BasicBlock : private LIR::Range
 
 #define BBF_COMPACT_UPD                                                                                                \
     (BBF_CHANGED | BBF_GC_SAFE_POINT | BBF_HAS_JMP | BBF_NEEDS_GCPOLL | BBF_HAS_IDX_LEN | BBF_BACKWARD_JUMP |          \
-     BBF_HAS_NEWARRAY | BBF_HAS_NEWOBJ)
+     BBF_HAS_NEWARRAY | BBF_HAS_NEWOBJ | BBF_HAS_NULLCHECK)
```

Comment:
Curious how you found these missing propagation bits... did you experiment with temporarily making `optEarlyPropFor...` always return true? (if not, may be worth a try).

Reply:
No, I just saw some null checks not getting removed in some diffs and tracked that down to the missing flags. Good idea to try with `optDoEarlyPropFor...` returning always true. Will try tomorrow.

Reply:
I added missing `OMF_HAS_NULLCHECK` and `BBF_HAS_NULLCHECK` in a couple of places but they didn't result in any new diffs. Changing `optDoEarlyPropFor...` to always return true does result in diff but they don't seem to be related to nullchecks. I will follow up outside of this PR.

Reply:
Apparently it was me who was forgetting to add these....